internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -182,6 +189,8 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 		return nil, err
 	}
 
+	det.SetState(details.StateWatching)
```

Comment:
Since the agent restarts essentially right after this, whether not we ever seen any of these transitions in Fleet depends on if a checkin happens to occur before we restart. 

We would be guaranteed to see this on the first checkin of the new agent version if it were persisted somewhere. I assume something like this will be done when we add support for the UPG_ROLLBACK state?

Although realistically if the upgrade completes quickly and without error then knowing anything about it isn't that interesting. It's the slow and failure cases that are interesting to see.

Reply:
Yeah, I haven't yet thought through the implementation of for `UPG_ROLLBACK` but I expect we'll use the upgrade marker file to persist this state so we can communicate it from the Upgrade Watcher process to the main Agent process.  We may want to expand on that idea and use the upgrade marker file to persist `UPG_WATCHING` as well — in fact, it will probably be the initial state in the upgrade marker file when it's created.

Longer term it would be nice to persist _all_ upgrade states in the upgrade marker file and use it to drive the upgrade state machine; that way if an Agent restarts anywhere in the middle of an upgrade, it should be able to pick up from where it left off (assuming every or almost every step of the upgrade process is idempotent).